test/src/apps/shm/memfd_create.c

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include "../test.h"
+
+#define FILE_SIZE 4096
+
+const char *msg1 = "jguk&auyg#eufg\n";
+const char *msg2 = "kug*skhikf%hsd\n";
+const char *msg_combined = "jguk&auyg#eufg\nkug*skhikf%hsd\n";
+
+FN_TEST(memfd_create)
```

Comment:
Why do writing a C test for memfd is preferred over reusing any tests in the gvisor or LTP test suites?